Week3/Day2/docs/Documentation.md

```diff
@@ -0,0 +1,290 @@
+  *.md
+schema/           # Folder containing different create sql queries.
+  *.sql    
+src/
+  pipeline/       # Folder containing python scripts    
+    *.py
+  sql/            # Folder containing DQL/DML.
+    *.sql
+```
+
+## 1. Let me explain about different sql files in schema.
+### i. `Schema/create_raw_employee_table.sql` file:
+``` 
+CREATE TABLE raw_employee(
+	employee_id VARCHAR(500),
+    first_name VARCHAR(500),
+    last_name VARCHAR(500),
+    department_id VARCHAR(500),
+    department_name VARCHAR(500),
+    manager_employee_id VARCHAR(500),
+    employee_role VARCHAR(500),
+    salary VARCHAR(500),
+    hire_date VARCHAR(500),
+    terminated_date VARCHAR(500),
+    terminated_reason VARCHAR(500),
+    dob VARCHAR(500),
+    fte VARCHAR(500),
+    location VARCHAR(500)
+);
+```
+Here I created a table `raw_employee` with necessary columns from dataset in data/.
+
+All entities are declared `VARCHAR(500)` because, for a bulk datas to extract, we may not delare exact datatypes which may cause problems later.
+For example: Format for date `2021-12-12` can come as `12-12-2021` and thus declaring datatypes may cause problem later. So, lets create talbe with all VARCHAR types.
+
+### ii. `schema/create_raw_timesheet_table.sql` file:
+```
+CREATE TABLE raw_timesheet(
+	employee_id VARCHAR(500),	
+	cost_center	VARCHAR(500),
+	punch_in_time VARCHAR(500),	
+	punch_out_time VARCHAR(500),	
+	punch_apply_date VARCHAR(500),	
+	hours_worked VARCHAR(500),	
+	paycode VARCHAR(500)
+);
+```
+This is SQL file to create `raw_timesheet` table according to dataset in data/.
+
+### iii.  `schema/create_employee_table_archieve.sql` file:
+```
+CREATE TABLE raw_employee_archieve(
+	employee_id VARCHAR(500),
+    first_name VARCHAR(500),
+    last_name VARCHAR(500),
+    department_id VARCHAR(500),
+    department_name VARCHAR(500),
+    manager_employee_id VARCHAR(500),
+    employee_role VARCHAR(500),
+    salary VARCHAR(500),
+    hire_date VARCHAR(500),
+    terminated_date VARCHAR(500),
+    terminated_reason VARCHAR(500),
+    dob VARCHAR(500),
+    fte VARCHAR(500),
+    location VARCHAR(500),
+    file_name VARCHAR(500)
+);
+```  
+This is SQL file to create `raw_employee_archieve` table to archieve `raw_employee` table after extraction.
+
+
+
+### iv.  `schema/create_timesheet_table_archieve.sql` file:
+```
+CREATE TABLE raw_timesheet_archieve(
+	employee_id VARCHAR(500),	
+	cost_center	VARCHAR(500),
+	punch_in_time VARCHAR(500),	
+	punch_out_time VARCHAR(500),	
+	punch_apply_date VARCHAR(500),	
+	hours_worked VARCHAR(500),	
+	paycode VARCHAR(500),
+    file_name VARCHAR(500)
+);
+```  
+This is SQL file to create `raw_timesheet_archieve` table to archieve `raw_timesheet` table after extraction.
+
+
+All entities are again declared VARCHAR for same problem above.
+
+
+## 2. `database_connection.py` file in  src/pipeline/ :
+
+We connect to Postgresql database regularly in each script through `psycohpg2` and use the cursor with connection.
```

Comment:
typo mistakes in many places

Reply:
Dai I corrected major typos in `Day2` branch for this file but was not reflected in this `Day 3` branch. (branch creation problem). I will resolve it while merging.